rclpy/test/test_destruction.py

```diff
@@ -162,11 +161,8 @@ def test_destroy_node_asap():
 
     try:
         node = rclpy.create_node('test_destroy_subscription_asap', context=context)
-        with node.handle:
-            node.destroy_node()
-            # handle valid because it's still being used
-            with node.handle:
-                pass
```

Comment:
@ahcorde why the change?

Reply:
> `# handle valid because it's still being used`

This was an intentional change in #739 with `Destroyable`. Once someone asks to destroy a `Destroyable`, it will block new uses of it. The `Handle` class allowed new uses as long as someone else was already using it, but that could lead to two threads blocking destruction forever as long as their uses overlapped.

Reply:
That makes total sense. I still think we should test that usage attempts after destruction requests fail.